### my first machine learning in Python
##### predicting types of iris

In [4]:
import sklearn

In [5]:
from sklearn.datasets import load_iris
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split

##### roading data-set

In [8]:
import pandas as pd

In [10]:
iris = load_iris()

iris_data = iris.data

In [12]:
iris_label = iris.target
print('iris target value:', iris_label)
print('iris target name:', iris.target_names)

iris target value: [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]
iris target name: ['setosa' 'versicolor' 'virginica']


##### spliting dataset into train & test set

In [13]:
X_train, X_test, y_train, y_test = train_test_split(iris_data, iris_label, test_size=0.2, random_state=11)

##### decision Tree Classifier

In [14]:
dt_clf = DecisionTreeClassifier(random_state=11)

In [15]:
dt_clf.fit(X_train, y_train)

DecisionTreeClassifier(random_state=11)

##### predicting 

In [16]:
pred = dt_clf.predict(X_test)

In [18]:
from sklearn.metrics import accuracy_score
print('accuracy of prediction: {0:.4f}'. format(accuracy_score(y_test, pred)))

accuracy of prediction: 0.9333


##### Output
###### prediction of tree algorism has 93.33% accuracy.

#### Summary
###### 1. spliting dataset: train & test set
###### 2. learning model: using ML algorithm based on learned data
###### 3. predicting: by learning model
###### 4. evaluating ML model